pkg/galera/config/config.go

```diff
@@ -70,27 +75,49 @@ wsrep_sst_auth="root:{{ .RootPassword }}"
 	if err != nil {
 		return nil, fmt.Errorf("error getting SST: %v", err)
 	}
+	wrappedPodIP, err := c.wrapIPAddress(podEnv.PodIP)
+	if err != nil {
+		return nil, fmt.Errorf("error wrapping address: %v", err)
+	}
+	gcommListenAddress, err := c.thisHostIP(podEnv.PodIP)
+	if err != nil {
+		return nil, fmt.Errorf("error getting address: %v", err)
+	}
+	gcommListenAddress, err = c.wrapIPAddress(gcommListenAddress)
+	if err != nil {
+		return nil, fmt.Errorf("error wrapping address: %v", err)
+	}
+	providerOpts := map[string]string{
+		"gmcast.listen_addr": fmt.Sprintf("tcp://%s:4567", gcommListenAddress),
+		"ist.recv_addr":      fmt.Sprintf("%s:4568", wrappedPodIP),
+	}
+	maps.Copy(providerOpts, galera.ProviderOptions)
```

Comment:
Thank you for review (I appreciate that especially because I am new to GO and operator framework). However, in this case it was intentional, I think user should have the last word about the configuration.
On the other hand when I am thinking more about it, I cann't imagine scenario for `ist.recv_add` that can be correctly set with `galera.ProviderOptions` because it depends on pod IP (I can imagine scenario when the provided default is not correct - e.g. in environment with multiple pod interfaces, but that is more complex to support and I think supporting it should not be part of this feature).

There is one more thing that for sure should be updated - documentation. There should be a note how to configure `wsrep_provider_options` (i.e. with `mdb.spec.galera.providerOptions` instead of putting it in custom config, because it will override defaults provided by the operator).

I will add `galera.ProviderOptions` validation to reject custom configuration of `ist.recv_add` (I think custom configuration of `gmcast.listen_addr` should not be denied)

Reply:
Fair enough, I agree with this rationale.

> There is one more thing that for sure should be updated - documentation. 

Regarding the documentation, I will take care of updating our current [Galera documentation](https://github.com/mariadb-operator/mariadb-operator/blob/main/docs/GALERA.md) to reflect what we mentioned here about the `mdb.spec.galera.providerOption`, most likely an advanced user will need this. Also, I think it will be worth adding a section dedicated to IPv6, as it is a common setup.

> I will add galera.ProviderOptions validation to reject custom configuration of ist.recv_add (I think custom configuration of gmcast.listen_addr should not be denied)

Great, you can extend the recover validation here which is used by the validation webhook:

https://github.com/mariadb-operator/mariadb-operator/blob/254fcad2ebef6f5b9b8b638b1ef321c22f3e5ce2/api/v1alpha1/mariadb_galera_types.go#L196

and maybe add some test cases [here](https://github.com/mariadb-operator/mariadb-operator/blob/main/api/v1alpha1/mariadb_galera_types_test.go#L239) if you feel it is relevant.

Reply:
Validation added, but in different location. I am not sure why you pointed to recovery validation, because from my understanding (correct me) this function should only validate `GaleraRecovery` struct, but `wsrepProvider` is not part of `GaleraRecovery` struct.

I've added validation to `api/v1alpha1/mariadb_webhook.go`, if that's not correct and it should be moved (probably together with `ReplicaThreads` validation), let me know.